pyEDAA/UCIS/CLI/__init__.py

```diff
@@ -0,0 +1,213 @@
+
+
+@export
+class Program(ProgramBase, ArgParseMixin):
+	"""Program class to implement the command line interface (CLI) using commands and options."""
+
+	def __init__(self) -> None:
+		super().__init__()
+
+		# Call the constructor of the ArgParseMixin
+		ArgParseMixin.__init__(
+			self,
+			prog="pyedaa-ucis",
+		  description=dedent('''\
+				'pyEDAA.UCIS Service Program' to query and transform data to/from UCIS to any other format.
+				'''),
+		  epilog=dedent("""\
+		    Currently the following output formats are supported:
+		     * Cobertura (statement coverage - Java oriented format)
+		  """),
+		  formatter_class=RawDescriptionHelpFormatter,
+		  add_help=False
+		)
+
+#	@CommonSwitchArgumentAttribute("-q", "--quiet",   dest="quiet",   help="Reduce messages to a minimum.")
+#	@CommonSwitchArgumentAttribute("-v", "--verbose", dest="verbose", help="Print out detailed messages.")
+#	@CommonSwitchArgumentAttribute("-d", "--debug",   dest="debug",   help="Enable debug mode.")
+	def Run(self) -> None:
+		ArgParseMixin.Run(self)
+
+	@DefaultAttribute()
+	def HandleDefault(self, _) -> None:
+		"""Handle program calls without any command."""
+		self._PrintHeadline()
+		self._PrintHelp()
+
+	@CommandAttribute("help", help="Display help page(s) for the given command name.", description="Display help page(s) for the given command name.")
+	@ArgumentAttribute(metavar="Command", dest="Command", type=str, nargs="?", help="Print help page(s) for a command.")
+	def HandleHelp(self, args) -> None:
+		"""Handle program calls with command ``help``."""
+		self._PrintHeadline()
+		self._PrintHelp(args.Command)
+
+	@CommandAttribute("version", help="Display version information.", description="Display version information.")
+	def HandleVersion(self, _) -> None:
+		"""Handle program calls with command ``version``."""
+		self._PrintHeadline()
+		self._PrintVersion()
+
+	@CommandAttribute("export", help="Export data from UCDB.", description="Export data from UCDB.")
+	@ArgumentAttribute("--ucdb",      metavar='UCDBFile',      dest="ucdb",      type=str, help="UCDB file in UCIS format (XML).")
+	@ArgumentAttribute("--cobertura", metavar='CoberturaFile', dest="cobertura", type=str, help="Cobertura code coverage file (XML).")
+	def HandleExport(self, args) -> None:
+		"""Handle program calls with command ``export``."""
+		self._PrintHeadline()
+
+		returnCode = 0
+		if args.ucdb is None:
+			print(f"Option '--ucdb <UCDBFile' is missing.")
+			returnCode = 3
+		if args.cobertura is None:
+			print(f"Option '--cobertura <CoberturaFile' is missing.")
+			returnCode = 3
+
+		if returnCode != 0:
+			exit(returnCode)
+
+		print(f"Exporting code coverage information from UCDB file to Cobertura format ...")
+
+		ucdbPath = Path(args.ucdb)
+		if not ucdbPath.exists():
+			raise FileNotFoundError(f"UCDB databse file '{ucdbPath}' not found.")
+
+		coberturaPath = Path(args.cobertura)
+
+		print(f"  IN  -> UCIS (XML):      {ucdbPath}")
```

Comment:
Can we extract the "in" and "out" formats from the order of the arguments? Currently we support UCIS -> Cobertura only. However, in the near future we might want to e.g. have Cobertura as input, or have other output formats. IMHO, we should account for such use cases since the begining.

Reply:
The current set of options are part of the `export` command. The reverse direction could use `import` or `merge` as main commands.